pkg/util/debug_logging.go

```diff
@@ -127,8 +135,40 @@ func (r *reducedRedudancyLogHook) Levels() []logrus.Level {
 // Private - Nil Logging Formatter
 // -----------------------------------------------------------------------------
 
-type nilFormatter struct{}
+// nilFormatter is a logrus.Formatter that allowed "flipping a switch" that will
+// cause the logger to stop emitting output.
+type nilFormatter struct {
```

Comment:
Just curious, why nilFormatter ? basically, it is log formatter.

Reply:
It's kinda a joke to indicate that it can cause log output to be empty (e.g. `nil`).